### ISSUE TITLE
Image Block: Media Sources Menu Inserted When Adding Caption

### ISSUE THREAD

**Issue by taipeicoder** [Automattic/jetpack#37943]
### Impacted plugin

Jetpack

### Quick summary

When adding a caption to an image, the media sources menu is unexpectedly inserted between the image and the caption. See screenshot for reference:

![Screenshot 2024-06-19 at 10 54 40 AM](https://github.com/Automattic/jetpack/assets/797888/2f0935e0-8c40-4ac2-a6b5-601d1b084216)

Or a recording:

https://github.com/Automattic/jetpack/assets/797888/d09bedaf-4533-429d-92e5-04838be1418f

Notes:
  - This also happens in the Gallery block.
  - Clicking on the menu items results in the media sources menu to be removed.
  - This happens in Gutenberg 18.5, downgrading to 18.4.1 and the issue is not reproducible.
  - This issue does not happen if Jetpack is disabled.

### Steps to reproduce

1. Create a new JN sites.
2. Install Gutenberg 18.5.
3. Connect Jetpack.
4. Head to the Site Editor, and add an image block.
5. Click on the Add caption button, and observe the issue described above.

### A clear and concise description of what you expected to happen.

For the image caption input field to be displayed.

### What actually happened

The image caption input field was displayed, but also the media sources menus is rendered along with it.

### Impact

Some (< 50%)

### Available workarounds?

No but the platform is still usable

### Platform (Simple and/or Atomic)

Atomic, Self-hosted

### Logs or notes

8351984-zd-a8c
8354098-zd-a8c
8354676-zd-a8c

**Comment by jeherve** [comment]
I'm thinking this may be related to the recent work on block bindings in core. @Automattic/frontity Is that something that's come up in your work in the past few weeks?

Those media sources are added by Jetpack by hooking into `blocks.registerBlockType`:
https://github.com/Automattic/jetpack/blob/9212d2fd350f61bb53d5337c5016372fd95432ad/projects/plugins/jetpack/extensions/shared/external-media/index.js#L9-L18

**Comment by SantosGuillamot** [comment]
> I'm thinking this may be related to the recent work on block bindings in core. Is that something that's come up in your work in the past few weeks?

I don't think any of the changes we added in that release should affect this functionality. We are working on the image caption now, but it hasn't been merged into Gutenberg yet.

I can try to help if I am able to reproduce the error in a local instance of the Gutenberg repo. Which is the best way to test this extension? Can I install it somehow to my local site?

**Comment by liviopv** [comment]
📌 REPRODUCTION RESULTS
- Tested on Simple – Replicated
- Tested on Atomic – Replicated
- Local site - No

📌 FINDINGS/SCREENSHOTS/VIDEO
Atomic with GB 18.5
https://github.com/Automattic/jetpack/assets/14153300/c9f29820-bbb8-44ec-b8ce-5eb1bce0e696

Atomic without GB 18.5 but with WordPress 6.6 beta 3
![CleanShot 2024-06-19 at 11 48 39@2x](https://github.com/Automattic/jetpack/assets/14153300/d1dc7d4f-e773-4fe0-92cc-bc68d98d3531)

Dotcom Simple
![CleanShot 2024-06-19 at 11 52 44@2x](https://github.com/Automattic/jetpack/assets/14153300/81a44d8a-7d5e-41de-ab8e-38639333da6c)

Local site with WordPress 6.6 Beta 3 + GB 18.5

https://github.com/Automattic/jetpack/assets/14153300/279ad7ee-df3c-4916-a3f4-452b813970a9

📌 ACTIONS
- Triaged

**Comment by jeherve** [comment]
Thanks for taking a look!

> I can try to help if I am able to reproduce the error in a local instance of the Gutenberg repo. Which is the best way to test this extension? Can I install it somehow to my local site?

Your local developement environment would need to be publicly available, so you can connect the Jetpack plugin to WordPress.com. Here in the Jetpack division we tend to rely on PCYsg-GJ2-p2 to make our local environments publicly available, but it does require a bit of setup.

> Local site - No

The Jetpack plugin doesn't seem to be activated on your local site, so you won't be able to replicate the problem since the External Media feature is only available on sites using the Jetpack plugin.

**Comment by liviopv** [comment]
> The Jetpack plugin doesn't seem to be activated on your local site, so you won't be able to replicate the problem since the External Media feature is only available on sites using the Jetpack plugin.

I know! That was just to confirm it was Jetpack-specific :)

**Comment by SantosGuillamot** [comment]
I tried to reproduce it using in Gutenberg (without JetPack) the same filters you shared, but I wasn't able to. I also tried setting up JetPack locally but, after tunneling it to my local domain, `wp-admin` is redirected so I can't connect it to WordPress.com.

By the way, these are [the changes introduced in Gutenberg 18.5](https://github.com/WordPress/gutenberg/releases/tag/v18.5.0). In case you are able to identify the issue.
